ironfish-rust/src/serializing/fr.rs

```diff
@@ -0,0 +1,109 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+use crate::errors::{IronfishError, IronfishErrorKind};
+use jubjub::Fr;
+
+use super::{bytes_to_hex, hex_to_bytes};
+
+pub const MAC_SIZE: usize = 16;
+
+pub trait FrSerializable {
+    fn serialize(&self) -> [u8; 32];
+    fn deserialize(bytes: [u8; 32]) -> Result<Fr, IronfishError>;
+    fn hex_key(&self) -> String;
+    fn from_hex(hex_key: &str) -> Result<Fr, IronfishError>;
+}
+
+impl FrSerializable for Fr {
```

Comment:
Can you provide context why we need this, with example code? Can't we just call `bytes_to_hex` on the `to_bytes` when we need it, This is not a key so we don't need to match the implementations like we did with proof generation key.

Reply:
This may be useful at some point, but I am not sure all this code merits one call, could be wrong

Reply:
The alternative would be to use the following logic in our napi layer everywhere we want to deserialize: 

<img width="1042" alt="image" src="https://github.com/iron-fish/ironfish/assets/13268167/e065f0ae-c851-46e2-8c66-641286ada199">

(idk if I'm explaining this right)

Writing tests will also be much easier

Reply:
Ok you convinced me :)